Modules/Scripted/DICOMLib/DICOMPlugin.py

```diff
@@ -277,3 +277,37 @@ def addSeriesInSubjectHierarchy(self,loadable,dataNode):
                               slicer.dicomDatabase.fileValue(firstFile, tags['studyTime']) )
         # Set item name
         shn.SetItemName(studyItemID, studyDescription + ' (' + studyDate + ')')
+
+  def mapSOPClassUIDToDICOMQuantityAndUnits(self, classUID):
+
+    MRname2UID = {
+        "MR Image Storage": "1.2.840.10008.5.1.4.1.1.4",
+        "Enhanced MR Image Storage": "1.2.840.10008.5.1.4.1.1.4.1",
+        "Legacy Converted Enhanced MR Image Storage": "1.2.840.10008.5.1.4.1.1.4.4"
+        }
+
+    CTname2UID = {
+        "CT Image Storage": "1.2.840.10008.5.1.4.1.1.2",
+        "Enhanced CT Image Storage": "1.2.840.10008.5.1.4.1.1.2.1",
+        "Legacy Converted Enhanced CT Image Storage": "1.2.840.10008.5.1.4.1.1.2.2"
+        }
+
+    quantity = None
+    units = None
+
+    # Note more specialized definitions can be specified for MR by more
+    # specialized plugins, see codes 110800 and on in
+    # http://dicom.nema.org/medical/dicom/current/output/chtml/part16/chapter_D.html
+    if classUID in MRname2UID.values():
+      quantity = slicer.vtkCodedEntry()
+      quantity.SetSchemeValueMeaning("DCM", "110852", "MR signal intensity")
+      units = slicer.vtkCodedEntry()
+      units.SetSchemeValueMeaning("UCUM", "1", "no units")
+
+    if classUID in CTname2UID.values():
+      quantity = slicer.vtkCodedEntry()
+      quantity.SetSchemeValueMeaning("DCM", "110851", "X-Ray Attenuation Coefficient")
```

Comment:
OK! (I just modified it because I've noticed that it was 110852 in the original pull request, probably copy-paste error)